criu/pie/restorer.c

```diff
@@ -340,6 +359,66 @@ static inline int restore_pdeath_sig(struct thread_restore_args *ta)
 	return 0;
 }
 
+/*
+ * Map an iovec with a given file,
+ * using `sys_mmap` instead of `sys_preadv`
+ * can accelerate restore process when
+ * there are some pages we won't touch later.
+ */
+static int map_iovec(struct iovec *iovs, loff_t off, int nr, struct task_restore_args *args)
+{
+	/* Using mmap to load content as an on-demand way rather than egarly read all the content */
+	int j;
+	VmaEntry *tmp_entry;
+	unsigned long iov_size, tmp;
+	loff_t local_offset = off;
+	ssize_t local_r, r = 0;
+	for (j = 0; j < nr; j++) {
+		void *p = (void *)iovs[j].iov_base;
```

Comment:
I don't like all these type conversions in this code. I think you can add the offset variable instead of p and it will work better:
```
unsigned long offset = 0;
unsigned long end = iovs[j].iov_len;
...
while (offset < end) {
....
    tmp = sys_mmap(iovs[j].iov_base + offset, iov_size, tmp_entry->prot | PROT_WRITE
    ...
```

Reply:
Ok, I'll change the type conversion.